File: test_ode_solver.py
#!/usr/bin/env python3
import math
import numpy as np
import matplotlib.pyplot as plt

from parameters import parameters
from exact_solution import t0, tf, solution, y_exact, y_prime, jacobian, compute_error_of_exact_solution
from ode_solver import ode_solver
from plot import plot_test

# set initial conditions
y0 = y_exact(t0)


# evolve ode system
y, t, dt, evaluations, reject = ode_solver(y0, t0, tf, y_prime, parameters, jacobian = jacobian)


# evaluate numerical accuracy
error = compute_error_of_exact_solution(t, y, y_exact)


# plot solution
log_plot = solution in ['exponential', 'inverse_power']

adaptive       = parameters['adaptive']
method         = parameters['method']
plot_variables = 2

plot_test(y, t, dt, t0, tf, method, adaptive, reject, evaluations, plot_variables, error, log_plot = log_plot)

File: julia/test_ode_solver.jl
#!/usr/bin/env julia
include("ode_solver.jl")
include("parameters.jl")
# using ExactSolution                   # how to do something like this?
include("exact_solution.jl")
using .ExactSolution 

v(x) = (println(x); x)	

y0 = y_exact(t0)

jacobian = 4

y, t, dt, evals, rejection = ode_solver(y0, t0, tf, y_prime, parameters, jacobian = jacobian)

v(y[end])
v(t[end])

# using Plots: plot, savefig
# fig = plot(t[:], y[1,:])                # basic figure 
# # @show fig                             # this didn't plt show after terminal
# savefig(fig, "plot.png")

